taichi/program/program.cpp

```diff
@@ -564,6 +564,22 @@ Ndarray *Program::create_ndarray(const DataType type,
   return ndarrays_.back().get();
 }
 
+Texture *Program::create_texture(const DataType type,
+                                 int num_channels,
+                                 const std::vector<int> &shape) {
```

Comment:
Slightly nit: pass in a copy of `shape`, then pad it to length 3 with `1`? This saves a few if-else checks.

Reply:
This is intended to support 1D/2D/3D images, so there will always be if checks i think...